Dockerfile

```diff
@@ -1,3 +1,11 @@
-FROM starefossen/github-pages:onbuild
+# Build the docs
+FROM starefossen/github-pages:147 AS build_docs
+COPY . /source
+RUN jekyll build --source /source --destination /site
 
-CMD bundle update && bundle exec jekyll serve -d /_site --watch -H 0.0.0.0 -P 4000
+# Reset with nginx, so we don't get docs source in the image
+FROM nginx:alpine
+# Override some nginx conf -- this gets added to default nginx conf
+COPY nginx-overrides.conf /etc/nginx/conf.d/000-default.conf
```

Comment:
Okay; found some issues; I forgot there's a default override file already in `/etc/nginx/conf.d/default.conf`. That one defines the "home" directory, but only for "localhost".

So, we should overwrite that file, and change the directory to copy the HTML to to `/usr/share/nginx/html`.

Here's a diff with the changes;

```patch
diff --git a/Dockerfile b/Dockerfile
index 8432103650..b1f497dcb6 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -6,6 +6,7 @@ RUN jekyll build --source /source --destination /site
 # Reset with nginx, so we don't get docs source in the image
 FROM nginx:alpine
 # Override some nginx conf -- this gets added to default nginx conf
-COPY nginx-overrides.conf /etc/nginx/conf.d/000-default.conf
-COPY --from=build_docs /site /etc/nginx/html/
-CMD echo -e "Docker docs are viewable at:\nhttp://0.0.0.0:4000"; exec nginx -g 'daemon off;'
\ No newline at end of file
+COPY nginx-overrides.conf /etc/nginx/conf.d/default.conf
+RUN rm /usr/share/nginx/html/*
+COPY --from=build_docs /site /usr/share/nginx/html
+CMD echo -e "Docker docs are viewable at:\nhttp://0.0.0.0:4000"; exec nginx -g 'daemon off;'
diff --git a/nginx-overrides.conf b/nginx-overrides.conf
index 55ab5d2e0a..e05e1edd8b 100644
--- a/nginx-overrides.conf
+++ b/nginx-overrides.conf
@@ -2,4 +2,9 @@ server {
   port_in_redirect off;
   listen 4000;
   error_page 403 404 /404.html;
-}
\ No newline at end of file
+
+  location / {
+    root   /usr/share/nginx/html;
+    index  index.html index.htm;
+  }
+}
```

Reply:
This also requires a change in #5367 and #5369 due to the paths changing.